File: 2013-04-19_simple_pub_sub_js/publish.js
var redis = require('redis');
var client = redis.createClient();
var params = { channel: process.argv[2], message: process.argv[3]};

client.on('ready', function() {

    if (params.channel && params.message) {
        var randKey = "Messages:" + (Math.random() * Math.random()).toString(16).replace('.', '');
        client.hset(randKey, "message", params.message);
        client.sadd('Channel:' + params.channel, randKey);
        client.publish(params.channel, params.message);
    }

    client.quit();
});

File: 2013-04-19_simple_pub_sub_js/publish.rb
require 'rubygems'
require 'redis'

channel = ARGV[0]
message = ARGV[1]
#puts "channel: #{channel}"
#puts "message: #{message}"

$redis = Redis.new
$redis.publish channel, message

rand_key = (Random.rand(12000) * Random.rand(12000)).to_i().to_s(16).sub('.', '')
$redis.hset(rand_key, "message", message)
$redis.sadd('Channel:' + channel, rand_key)

